### PR TITLE
699 social group definition

### PR DESCRIPTION
**Pull request by hwindsor**
## Pull Request Overview

- Social group role setting via new site settings API,
- Moved variable specific houston config into a table rather than hard coded in the resources files.

---
```
[POST] /api/v1/site-settings/main/social_group_roles
{
    '_value': '',  [
        {'label': 'Matriarch', 'multipleInGroup': False},
        {'label: 'IrritatingGit', 'multipleInGroup': True},
    }
}
```

### CODE REVIEW

**Comment by brmscheiner** [comment]
It would look more like other similar APIs if _value was something like

```
[
{ label: 'Matriarch', value: <guid>, multipleInGroup: false },
{ label: 'IrritatingGit', value: <guid>, multipleInGroup: true },
]
```

But if it has to look like this for some particular reason I can still work with it

**Review by colinwkingen — APPROVED** [review]
LGTM

**Comment by hwindsor** [comment]
> It would look more like other similar APIs if _value was something like
> 
> ```
> [
> { label: 'Matriarch', value: <guid>, multipleInGroup: false },
> { label: 'IrritatingGit', value: <guid>, multipleInGroup: true },
> ]
> ```
> 
> But if it has to look like this for some particular reason I can still work with it

I have no problem with changing it to that. I chose the _value construct as that was how other settings were done. 
That was what I found in the frontend when I looked too but happy to make it as above still

**Comment by brmscheiner** [comment]
@hwindsor well it would still be 
```
_value: [
  { label: 'Matriarch', value: <guid>, multipleInGroup: false },
  { label: 'IrritatingGit', value: <guid>, multipleInGroup: true },
]
```
right?

**Comment by hwindsor** [comment]
> @hwindsor well it would still be
> 
> ```
> _value: [
>   { label: 'Matriarch', value: <guid>, multipleInGroup: false },
>   { label: 'IrritatingGit', value: <guid>, multipleInGroup: true },
> ]
> ```
> 
> right?
@brmscheiner 
Ah OK Yes I see what you mean. Yes I can make it like that.

File: app/modules/site_settings/models.py [hunk 1]
```diff
@@ -106,17 +176,18 @@ def set_key_value(cls, key, value):
...
-            msg = f'Houston Setting key={key}, value incorrect type value={value}'
+            msg = f'Houston Setting key={key}, value incorrect type value={value},'
+            msg += f'needs to be {cls.HOUSTON_SETTINGS[key]["type"]}'
             raise HoustonException(log, msg)
 
         if isinstance(value, str):
             log.debug(f'updating Houston Setting key={key}')
             cls.set(key, string=value, public=cls.HOUSTON_SETTINGS[key]['public'])
-        elif isinstance(value, dict):
+        elif isinstance(value, dict) or isinstance(value, list):
```
Comment (hwindsor) [inline app/modules/site_settings/models.py:186]: @naknomum I really want your opinion on this change. What was there just looked wrong. Please comment

**Pull request merged**